### ISSUE TITLE
[Bug]: Button click causes error: Runtime.callFunctionOn timed out. Increase the 'protocolTimeout'

### ISSUE THREAD

**Issue by cory-weiner** [puppeteer/puppeteer#12723]
### Minimal, reproducible example

```TypeScript
I am using the pre-build docker image. Upon clicking a button on the page, the script seems to stop and eventually throws the error:

"Runtime.callFunctionOn timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed."

I have tested this with image versions 22, 21, and 20:
ghcr.io/puppeteer/puppeteer:22

Importantly, this issue only occurs when running inside the docker container. When running the exact same code on my local machine, clicking the button works fine.

It might be worth noting, I'm using Apple Silicon M3 Max chip.

I've tried clicking the button multiple ways, including:

const button = await page.$('.downloadButton');
await button.click()

I've also tried clicking the button inside of evaluate function. I can confirm in both instances the button element exists on the page and the sector finds the correct element.

   await page.evaluate(() => {
        let button = document.querySelector('.downloadButton')
        button.dispatchEvent(new MouseEvent('click', { 'bubbles': true }))
    })

   await page.evaluate(() => {
        let button = document.querySelector('.downloadButton')
        button.click()
    })
```

### Error string

Runtime.callFunctionOn timed out. Increase the 'protocolTimeout' setting in launch/connect calls for a higher timeout if needed.

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Expectation

I would expect the button to be clicked

### Reality

The protocolTimeout error is thrown

### Puppeteer version

22.12.1

### Node version

20.9.0

### Package manager

npm

### Package manager version

10.8.1

### Operating system

macOS

**Comment by OrKoN** [comment]
Could you please provide a complete executable script (perhaps as a Dockerfile)? including a test site/page (URL) that demonstrates the issue.

**Comment by cory-weiner** [comment]
I will try to replicate it. Unfortunately the button in question is behind an authenticated page.

Worth noting, I've just discovered the issue does NOT happen when running the container on AWS ECS. It only appears to happen when running the via docker on my machine. If I had to guess it is related to the Apple Silicon chip but perhaps another apple user can verify. I'll try to put together a replicable test case.
